src/macaron/slsa_analyzer/analyzer.py

```diff
@@ -540,6 +541,14 @@ def to_analysis_target(config: Configuration, available_domains: list[str]) -> A
             except ValueError as error:
                 raise InvalidPURLError(f"Invalid input PURL: {config.get_value('purl')}") from error
 
+        # Validate the purl object, which is user-controllable.
+        if parsed_purl:
+            purl_str = parsed_purl.to_string()
+            # The urlparse API sanitizes unsafe characters.
+            parsed_url = urllib.parse.urlparse(purl_str)
+            if purl_str != parsed_url.geturl():
+                raise ValueError("The PURL provided as input has unsafe characters.")
```

Comment:
This case raises `ValueError`, but `ValueError` is neither documented in the method docstring nor caught at the call site of this method.
I also wonder if `ValueError` here should be converted to `InvalidPURLError` to be consistent with the rest of the method, or should it be a separate error type to be handled differently from the rest.